KsApi/MockService.swift

```diff
@@ -514,6 +515,19 @@ internal struct MockService: ServiceType {
     )
   }
 
+  func fetchGraphCategories(query: NonEmptySet<Query>) -> SignalProducer<RootCategoriesEnvelope, GraphError> {
+    return SignalProducer(value: self.fetchGraphCategoriesResponse!)
```

Comment:
I think it'd be better to do an `if let response = self.fetchGraphCategoriesResponse { ... }` and if it's `nil` just return `.empty`. Better to have a failing test than tests crashing altogether because of potentially unwrapping a `nil` value.

Reply:
In this case I'm unwrapping the optional because there is no way for it to be nil, since it will returns a template if the value of `fetchGraphCategoriesResponse` is nil. But I agree that it will make the test code more reliable adding a `if` statement.